tests/compiler/std/string.ts

```diff
@@ -344,6 +344,150 @@ assert(dtoa(0.000035689) == "0.000035689");
 // assert(dtoa(f32.MAX_VALUE) == "3.4028234663852886e+38"); // FIXME
 // assert(dtoa(f32.EPSILON) == "1.1920928955078125e-7"); // FIXME
 
+
+// toUpperCase
+assert("abcdefghijklmnopqrstuvwxyz".toUpperCase() == "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "-1");
+assert("aBcDeFgHiJkLmNoPqRsTuVwXyZ".toUpperCase() == "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "-2");
+assert("AbCdEfGhIjKlMnOpQrStUvWxYz".toUpperCase() == "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "-3");
+
+// Examples generated by @MaxGraey
```

Comment:
I guess this comment unnecessary)

Reply:
I removed it, but it was just a comment so I could remember to visit them. If you have any more cases, please let me know. I will make this function as robust as I can.

```ts
store<u16>(out + outpos, 0xD801); // always store 0xD801
store<u16>(out + outpos + 2, char2 - 40);

// trying
store<u32>(out + outpos,
  (<u32>(char2 - 40) << <u32>16) & <u32>0xD801); // always store 0xD801(number - 40)
```

I'm going to try and turn this into a single store instruction, but this seems to error. Any help?

Reply:
It should be `((char2 - 40) << 16) | 0xD801`

Reply:
Never mind. It needs a `|` not an `&`.